core/templates/pages/exploration-editor-page/settings-tab/settings-tab.component.html

```diff
@@ -36,16 +36,16 @@ <h3 class="e2e-test-settings-container" tabindex="0">Basic Settings</h3>
                   <p>
                     <mat-form-field class="e2e-test-exploration-category" appearance="fill">
                       <mat-label for="explorationCategory" class="d-block">Add a category (Choose or type new)</mat-label>
-                      <mat-select [(value)]="explorationCategoryService.displayed" class="e2e-test-exploration-category-dropdown">
+                      <mat-select [(value)]="explorationCategoryService.displayed"
+                                  (valueChange)="updateCategoryListWithUserData()"
```

Comment:
It took me some time to understand this logic, it looks very indirect. Maybe updateCategoryListWithUserData should take an argument - what value to set? Currently we set value for the newCategory only in the filterChoices which is super unexpected.

Reply:
The value only changes when the user selects a certain element in the select so it makes sense here IG.